src/osdc/Journaler.cc

```diff
@@ -640,7 +640,7 @@ void Journaler::_do_flush(unsigned amount)
       waiting_for_zero = true;
       return;
     }
-    if (newlen < len) {
+    if ((uint64_t)newlen < len) { //here newlen is a positive value.
```

Comment:
Can we use "static_cast<uint64_t>(newlen)" here please instead of the c-style cast? Please also remove the comment, it shouldn't really be needed.